### PR TITLE
[PORT #6366] set default PoolRouter SupervisorStrategy to Restart

### LINKED ISSUE #6295: Pool routers use `Directive.Escalate` by default when supervising routees - this causes unintended consequences
**Version Information**
Version of Akka.NET? 1.4.46 and earlier, 1.5.0-alpha3 and earlier.
Which Akka.NET Modules? Akka

**Describe the bug**

Currently, the default `SupervisionStrategy` on all `Pool` routers uses `Directive.Escalate`, which causes the router to restart and kill all routees every time any of the routees restart. This can be problematic, as it means that all routees are restarted even if only one of them has failed.

Instead, I suggest changing the default `SupervisionStrategy` to return a `Directive.Restart `instead. This would cause only the failed routee to be restarted, rather than restarting all routees in the pool. This would allow for more granular control over how failures are handled, and can prevent unnecessary restarts of healthy routees.

**To Reproduce**

Reproduced with one of my unit tests on https://github.com/akkadotnet/akka.net/pull/6294

```csharp
// unit test to assert that a pool router actor's children generate telemetry events, but not the router actor itself
[Fact]
public async Task ActorTelemetry_must_be_accurate_for_pool_router()
{
    // create a TelemetrySubscriber actor
    var subscriber = Sys.ActorOf(Props.Create<TelemetrySubscriber>(), "subscriber");
    
    // request current telemetry values (ensure that the actor has started, so counter values will be accurate)
    await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
    
    // create a pool router
    var router = Sys.ActorOf(Props.Create<ChildActor>().WithRouter(new RoundRobinPool(10)), "router");

    // awaitassert collecting data from the telemetry subscriber until we can see that 10 actors have been created
    await AwaitAssertAsync(async () =>
    {
        var telemetry = await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
        Assert.Equal(12, telemetry.ActorCreated);
        // assert no restarts or stops recorded
        Assert.Equal(0, telemetry.ActorRestarted);
        Assert.Equal(0, telemetry.ActorStopped);
    }, RemainingOrDefault);
    
    // send a message to the router to restart all children
    router.Tell(new Broadcast(RestartChildren.Instance));

    // await assert collecting data from the telemetry subscriber until we can see that 10 actors have been restarted
    await AwaitAssertAsync(async () =>
    {
        var telemetry = await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
        // assert that actor start count is still 10
        Assert.Equal(12, telemetry.ActorCreated);
        Assert.Equal(10, telemetry.ActorRestarted);
        // assert no stops recorded
        Assert.Equal(0, telemetry.ActorStopped);
    }, RemainingOrDefault);
    
    // GracefulStop router actor and assert that 10 actors have been stopped
    await router.GracefulStop(RemainingOrDefault);
    await AwaitAssertAsync(async () =>
    {
        var telemetry = await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
        // assert that actor start count is still 10
        Assert.Equal(12, telemetry.ActorCreated);
        Assert.Equal(10, telemetry.ActorRestarted);
        Assert.Equal(10, telemetry.ActorStopped);
    }, RemainingOrDefault);
}
```

**Expected behavior**

Each child should be restarted individually and independently from each other.

**Actual behavior**

`Assert.Equal(10, telemetry.ActorRestarted);` should be true, but instead the `telemetry.ActorRestarted` value is 110!

That's because the following line:

```csharp
// send a message to the router to restart all children
  router.Tell(new Broadcast(RestartChildren.Instance));
```

Causes the router to restart all of its children 10 times apiece.

### PR DESCRIPTION
**Pull request by Arkatufus**
Port of #6366

* close #6295 - set default PoolRouter SupervisorStrategy to Restart
* use the `SupervisionStrategy.DefaultStrategy`
* Fix unit test
* Change unit test to use RoundRobinPool router

(cherry-picked from ffd9a9e15e2841cabb7ac9e71db937c7ea320cbd)

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Ty!

File: src/core/Akka.Tests/Actor/ActorTelemetrySpecs.cs [hunk 1]
```diff
@@ -296,7 +295,7 @@ public async Task ActorTelemetry_must_be_accurate_for_pool_router()
                 var telemetry = await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
                 // assert that actor start count is still 10
                 Assert.Equal(12, telemetry.ActorCreated);
-                Assert.Equal(110, telemetry.ActorRestarted);
+                Assert.Equal(10, telemetry.ActorRestarted);
```
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Actor/ActorTelemetrySpecs.cs:298]: We will also need to backport this to v1.4 as well

Comment (Arkatufus) [reply src/core/Akka.Tests/Actor/ActorTelemetrySpecs.cs:298]: I wonder why we didn't caught this before... I guess we missed it amongst the racy unit test noise?

Comment (Arkatufus) [reply src/core/Akka.Tests/Actor/ActorTelemetrySpecs.cs:298]: added the unit test fix in #6376

**Pull request merged**